webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-grading.js

```diff
@@ -345,35 +346,69 @@ export class SakaiRubricGrading extends RubricsElement {
       headers: { "authorization": this.token }
     }).done(rubric => {
+	
+      if (!this.isPeerOrSelf) {
 
-      $.ajax({
-        url: `/rubrics-service/rest/evaluations/search/by-tool-item-and-associated-item-and-evaluated-item-ids?toolId=${this.toolId}&itemId=${this.entityId}&evaluatedItemId=${this.evaluatedItemId}`,
-        headers: { "authorization": this.token }
-      }).done(data => {
+        $.ajax({
+          url: `/rubrics-service/rest/evaluations/search/by-tool-item-and-associated-item-and-evaluated-item-ids?toolId=${this.toolId}&itemId=${this.entityId}&evaluatedItemId=${this.evaluatedItemId}`,
+          headers: { "authorization": this.token }
+        }).done(data => {
 
-        this.evaluation = data._embedded.evaluations[0] || { criterionOutcomes: [] };
-        this.selectedRatings = this.evaluation.criterionOutcomes.map(ed => ed.selectedRatingId);
-        if (this.criteria) this.decorateCriteria();
-        this.existingEvaluation = true;
+          this.evaluation = data._embedded.evaluations[0] || { criterionOutcomes: [] };
+          this.selectedRatings = this.evaluation.criterionOutcomes.map(ed => ed.selectedRatingId);
+          if (this.criteria) this.decorateCriteria();
+          this.existingEvaluation = true;
 
-        this.rubric = rubric;
+          this.rubric = rubric;
 
-        this.criteria = this.rubric.criterions;
-        this.criteria.forEach(c => {
+          this.criteria = this.rubric.criterions;
+          this.criteria.forEach(c => {
 
           if (!c.selectedvalue) {
             c.selectedvalue = 0;
           }
           c.pointrange = this.getHighLow(c.ratings, "points");
+          });
+
+          this.decorateCriteria();
+        }).fail((jqXHR, textStatus, errorThrown) => {
+          console.log(textStatus);console.log(errorThrown);
         });
 
-        this.decorateCriteria();
-      }).fail((jqXHR, textStatus, errorThrown) => {
-        console.log(textStatus);console.log(errorThrown);
-      });
+      } else {
+	
+        $.ajax({
+          url: `/rubrics-service/rest/evaluations/search/peer-by-tool-item-and-associated-item-and-evaluated-item-ids?toolId=${this.toolId}&itemId=${this.entityId}&evaluatedItemId=${this.evaluatedItemId}`,
+          headers: { "authorization": this.token }
+        }).done(data => {
+
+          this.evaluation = data._embedded.evaluations[0] || { criterionOutcomes: [] };
+          this.selectedRatings = this.evaluation.criterionOutcomes.map(ed => ed.selectedRatingId);
+          if (this.criteria) this.decorateCriteria();
+          this.existingEvaluation = true;
+
+          this.rubric = rubric;
+
+          this.criteria = this.rubric.criterions;
+          this.criteria.forEach(c => {
+
+            if (!c.selectedvalue) {
+              c.selectedvalue = 0;
+            }
+            c.pointrange = this.getHighLow(c.ratings, "points");
+          });
+
+            this.decorateCriteria();
```

Comment:
You should extract all this criteria decoration code into another function. It's the same code. Dry it out :)

Reply:
Done.